src/appservice/config/config.ts

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as fs from "fs";
+import { load } from "js-yaml";
+
+export interface IConfig {
+    /** Details for the homeserver the appservice will be serving */
+    homeserver: {
+        /** The domain of the homeserver that is found at the end of mxids */
+        domain: string,
+        /** The url to use to acccess the client server api */
+        url: string
+    },
+    /** Details for the database backend */
+    db: {
+        /** Postgres connection string  */
+        connectionString: string
+    },
+    /** Config for the web api used to access the appservice via the widget */
+    webAPI: {
+        port: number
```

Comment:
I don't see any port number for the appservice API itself, is that normal?

Reply:
Yes, that's provided by the cli when it is started. (Though that is a restriction imposed by matrix-appservice-bridge when using their cli i think, i don't know the reason why though)